src/lib/tuning/index/base_index_evaluator.cpp

```diff
@@ -0,0 +1,222 @@
+#include "base_index_evaluator.hpp"
+
+#include <algorithm>
+#include <list>
+#include <memory>
+#include <string>
+
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "operators/get_table.hpp"
+#include "operators/table_scan.hpp"
+#include "operators/validate.hpp"
+#include "optimizer/column_statistics.hpp"
+#include "optimizer/table_statistics.hpp"
+#include "sql/gdfs_cache.hpp"
+#include "sql/sql_query_cache.hpp"
+#include "storage/index/base_index.hpp"
+#include "storage/storage_manager.hpp"
+#include "types.hpp"
+#include "utils/logging.hpp"
+
+namespace opossum {
+
+BaseIndexEvaluator::BaseIndexEvaluator() {}
+
+void BaseIndexEvaluator::evaluate(std::vector<std::shared_ptr<TuningChoice>>& choices) {
+  // Allow concrete implementation to initialize
+  _setup();
+
+  // Scan query cache for indexable table column accesses
+  _inspect_query_cache();
```

Comment:
For me, seeing a method without a return type and unknown side effects is little better than spaghetti code. Scrolling over that method it looks like it sets `_access_records`. Why not do `auto access_records = _inspect_query_cache`?

Reply:
We modified the methods to have explicit arguments and return values instead of using the class members.